src/zenml/zen_server/auth.py

```diff
@@ -1045,56 +1046,85 @@ def generate_access_token(
     )
 
 
-def generate_artifact_download_token(artifact_version_id: UUID) -> str:
-    """Generate a JWT token for artifact download.
+def generate_download_token(
+    download_type: DownloadType,
+    resource_id: UUID,
+    extra_data: Optional[Dict[str, Any]] = None,
+    expires_in_seconds: int = 30,
+) -> str:
+    """Generate a JWT token for downloading content.
 
     Args:
-        artifact_version_id: The ID of the artifact version to download.
+        download_type: The type of content being downloaded.
+        resource_id: The ID of the resource to download.
+        extra_data: Optional extra data to include in the token.
+        expires_in_seconds: Token expiration time in seconds.
 
     Returns:
-        The JWT token for the artifact download.
+        The JWT token for the download.
     """
     import jwt
 
     config = server_config()
 
+    payload = {
+        "exp": utc_now() + timedelta(seconds=expires_in_seconds),
+        "download_type": download_type.value,
+        "resource_id": str(resource_id),
+    }
+
+    if extra_data:
+        payload.update(extra_data)
+
     return jwt.encode(
-        {
-            "exp": utc_now() + timedelta(seconds=30),
-            "artifact_version_id": str(artifact_version_id),
-        },
+        payload,
         key=config.jwt_secret_key,
         algorithm=config.jwt_token_algorithm,
     )
 
 
-def verify_artifact_download_token(
-    token: str, artifact_version_id: UUID
+def verify_download_token(
+    token: str,
+    download_type: DownloadType,
+    resource_id: UUID,
+    extra_data: Optional[Dict[str, Any]] = None,
```

Comment:
The name seems a little off IMO, maybe `extra_claims`?

Reply:
Renamed it.